script/trading/seaport/DeployImmutableSignedZoneV2.s.sol

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Immutable Pty Ltd 2018 - 2024
+// SPDX-License-Identifier: Apache-2
+// solhint-disable-next-line compiler-version
+pragma solidity 0.8.20;
+
+import "forge-std/Test.sol";
+import {ImmutableSignedZoneV2} from "../../../contracts/trading/seaport/zones/immutable-signed-zone/v2/ImmutableSignedZoneV2.sol";
+
+/**
+ * @title IDeployer Interface
+ * @notice This interface defines the contract responsible for deploying and optionally initializing new contracts
+ *  via a specified deployment method.
+ * @dev Credit to axelarnetwork https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/interfaces/IDeployer.sol
+ */
+interface IDeployer {
+    function deploy(bytes memory bytecode, bytes32 salt) external payable returns (address deployedAddress_);
+}
+
+struct DeploymentArgs {
+    address signer;
+    address factory;
+    string salt;
+}
+
+struct ZoneDeploymentArgs {
+    address owner;
+    string name;
+    string apiEndpoint;
+    string documentationURI;
+}
+
+contract DeployImmutableSignedZoneV2 is Test {
+    function testDeploy() external {
+        /// @dev Fork the Immutable zkEVM testnet for this test
+        string memory rpcURL = "https://rpc.testnet.immutable.com";
+        vm.createSelectFork(rpcURL);
+
+        /// @dev These are Immutable zkEVM testnet values where necessary
+        DeploymentArgs memory deploymentArgs = DeploymentArgs({
+            signer: 0xdDA0d9448Ebe3eA43aFecE5Fa6401F5795c19333,
+            factory: 0x37a59A845Bb6eD2034098af8738fbFFB9D589610,
+            salt: "salty"
+        });
+
+        /// @dev These are Immutable zkEVM testnet values where necessary
+        ZoneDeploymentArgs memory zoneDeploymentArgs = ZoneDeploymentArgs({
+            owner: address(0xC606830D8341bc9F5F5Dd7615E9313d2655B505D),
+            name: "TestImmutableSignedZoneV2",
+            apiEndpoint: "https://api.sandbox.immutable.com/",
```

Comment:
In the V1 zone deployment script, we've left the `api_endpoint` empty. Having the api endpoint info is fine but the script will need to account for the different values in different evnironments.

Reply:
@naveen-imtb this is the test, actual deployments will use env var during deployment